apiserver/facades/client/machinemanager/machinemanager.go

```diff
@@ -393,6 +436,19 @@ func (mm *MachineManagerAPI) destroyMachine(args params.Entities, force, keep bo
 				return fail(err)
 			}
 		}
+
+		// Ensure that when the machine has been removed that all the leadership
+		// references to that machine are also cleared up.
+		results, err := mm.leadership.UnpinApplicationLeadersByName(machineTag, applicationNames)
```

Comment:
Not sure we can return a failure here.
The machine will have been removed, and we will tell the caller that we failed to remove it.

Reply:
@manadart that's what I was wondering, didn't get to ask yesterday because of the commotion.